src/charm.py

```diff
@@ -98,9 +128,14 @@ def _on_config_changed(self, event):
         """Handle changed configuration."""
         self.publish_config()
 
-    def publish_config(self):
+    def _on_server_cert_changed(self, _):
+        self._update_cert()
+        self._on_nginx_pebble_ready(_)
```

Comment:
I don't like calling a hook callback here directly. Maybe make a function called self.configure() or something and call that in both hooks.